src/backend/src/routes/work-packages.routes.ts

```diff
@@ -14,46 +14,44 @@ workPackagesRouter.get('/', getAllWorkPackages);
 workPackagesRouter.get('/:wbsNum', getSingleWorkPackage);
 workPackagesRouter.post(
   '/create',
-  body('userId').isInt({ min: 0 }),
-  body('crId').isInt({ min: 0 }),
-  body('name').not().isEmpty(),
-  body('projectWbsNum.carNumber').isInt({ min: 0 }),
-  body('projectWbsNum.projectNumber').isInt({ min: 0 }),
-  body('projectWbsNum.workPackageNumber').isInt({ min: 0 }),
+  body('userId').isInt({ min: 0 }).not().isString(),
+  body('crId').isInt({ min: 0 }).not().isString(),
+  body('name').isString().not().isEmpty(),
+  body('projectWbsNum.carNumber').isInt({ min: 0 }).not().isString(),
+  body('projectWbsNum.projectNumber').isInt({ min: 0 }).not().isString(),
+  body('projectWbsNum.workPackageNumber').isInt({ min: 0 }).not().isString(),
   body('startDate').isDate(),
-  body('duration').isInt({ min: 0 }),
-  body('dependencies.*.carNumber').isInt({ min: 0 }),
-  body('dependencies.*.projectNumber').isInt({ min: 0 }),
-  body('dependencies.*.workPackageNumber').isInt({ min: 0 }),
+  body('duration').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.carNumber').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.projectNumber').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.workPackageNumber').isInt({ min: 0 }).not().isString(),
   body('expectedActivities').isArray(),
-  body('expectedActivities.*').isString(),
-  body('expectedActivities.*').not().isEmpty(),
+  body('expectedActivities.*').isString().not().isEmpty(),
   body('deliverables').isArray(),
-  body('deliverables.*').isString(),
-  body('deliverables.*').not().isEmpty(),
+  body('deliverables.*').isString().not().isEmpty(),
   createWorkPackage
 );
 workPackagesRouter.post(
   '/edit',
-  body('workPackageId').isInt({ min: 0 }),
-  body('userId').isInt({ min: 0 }),
-  body('crId').isInt({ min: 0 }),
-  body('name').not().isEmpty(),
+  body('workPackageId').isInt({ min: 0 }).not().isString(),
+  body('userId').isInt({ min: 0 }).not().isString(),
+  body('crId').isInt({ min: 0 }).not().isString(),
+  body('name').isString().not().isEmpty(),
   body('startDate').isDate(),
-  body('duration').isInt({ min: 0 }),
-  body('dependencies.*.carNumber').isInt({ min: 0 }),
-  body('dependencies.*.projectNumber').isInt({ min: 0 }),
-  body('dependencies.*.workPackageNumber').isInt({ min: 0 }),
+  body('duration').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.carNumber').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.projectNumber').isInt({ min: 0 }).not().isString(),
+  body('dependencies.*.workPackageNumber').isInt({ min: 0 }).not().isString(),
   body('expectedActivities').isArray(),
-  body('expectedActivities.*.id').isInt({ min: 0 }),
-  body('expectedActivities.*.detail').not().isEmpty(),
+  body('expectedActivities.*.id').isInt({ min: -1 }).not().isString(),
```

Comment:
Why are only these two ids allowed to be -1?

Reply:
lmao it's a weird way that I impleented the work package edit endpoint where if the id is -1 that means it's a new bullet being added.

Definitely something that we will address in a future ticket is standardizing the project and work package edit endpoints and combining their utils functions